nixos/doc/manual/installation/installing-usb.section.md

```diff
@@ -1,31 +1,49 @@
-# Booting from a USB Drive {#sec-booting-from-usb}
+# Booting from a USB flash drive {#sec-booting-from-usb}
 
-For systems without CD drive, the NixOS live CD can be booted from a USB
-stick. You can use the `dd` utility to write the image:
-`dd if=path-to-image of=/dev/sdX`. Be careful about specifying the correct
-drive; you can use the `lsblk` command to get a list of block devices.
+The image has to be written verbatim to the USB flash drive for it to be
+bootable on UEFI and BIOS systems. Here are the recommended tools to do that.
 
-::: {.note}
-::: {.title}
-On macOS
+## Creating bootable USB flash drive with a graphical tool {#sec-booting-from-usb-graphical}
+
+Etcher is a popular and user-friendly tool. It works on Linux, Windows and macOS.
+
+Download it from [balena.io](https://www.balena.io/etcher/), start the program,
+select the downloaded NixOS ISO, then select the USB flash drive and flash it.
+
+::: {.warning}
+Etcher reports errors and usage statistics by default, which can be disabled in
+the settings. Some users might not like that.
 :::
 
+An alternative is [USBImager](https://gitlab.com/bztsrc/usbimager/#usbimager),
+which is very simple and does not connect to the internet. Download the version
+with write-only (wo) interface for your system. Start the program,
+select the image, select the USB flash drive and click "Write".
+
+## Creating bootable USB flash drive from Terminal on Linux {#sec-booting-from-usb-linux}
+
+Use the `dd` utility to write the image to the USB flash drive:
+
 ```ShellSession
-$ diskutil list
-[..]
-/dev/diskN (external, physical):
-   #:                       TYPE NAME                    SIZE       IDENTIFIER
-[..]
-$ diskutil unmountDisk diskN
-Unmount of all volumes on diskN was successful
-$ sudo dd if=nix.iso of=/dev/rdiskN bs=1M
+sudo dd if=<path-to-image> of=/dev/sdX bs=4M conv=fsync oflag=direct status=progress
 ```
 
-Using the \'raw\' `rdiskN` device instead of `diskN` completes in
+Be careful to replace `sdX` with the correct device! You can use the `lsblk`
+command to get a list of block devices and their sizes.
+
+## Creating bootable USB flash drive from Terminal on macOS {#sec-booting-from-usb-macos}
+
+Use the `dd` utility to write the image to the USB flash drive. Unmount the disk before.
+
+```ShellSession
+diskutil unmountDisk diskX
+sudo dd if=<path-to-image> of=/dev/rdiskX bs=1m
```

Comment:
is `1m` right? the case may be important with this `dd`.

And shouldn't we recommend the same block size on all platforms?

Reply:
`1m` is correct for macOS dd. dd from gnu uses `1M`

i did not change that command, so it should work well. not sure if we can safely remove the `bs`

Reply:
dd defaults to bs=512, which is really slow:

```sh
# transfer 1GB with bs=4m
% time dd if=/dev/zero of=/dev/null bs=4194304 count=256
256+0 records in
256+0 records out
1073741824 bytes transferred in 0.043489 secs (24689975261 bytes/sec)
dd if=/dev/zero of=/dev/null bs=4194304 count=256
0.00s user 0.04s system 96% cpu 0.046 total

# transfer 1GB with bs=512 (the default)
% time dd if=/dev/zero of=/dev/null bs=512 count=2097152
2097152+0 records in
2097152+0 records out
1073741824 bytes transferred in 16.644957 secs (64508538 bytes/sec)
dd if=/dev/zero of=/dev/null bs=512 count=2097152 
0.91s user 13.67s system 87% cpu 16.648 total
```

So on this hypothetical zero-latency disk, bs=4M produces around a 350X speedup. In the real world I've seen 10X faster.

If the issue is Gnu vs. BSD suffixes, can you use `bs=4194304`?

Reply:
Thanks for the explanation! Then we should use 4m for both.

We have different commands for GNU and BSD, so we can use the short version, they only have to be correct.